frontend/src/pages/GroupPage.jsx

```diff
@@ -1,15 +1,152 @@
 import Template from "../components/template";
+import ModalHeader from "../components/modalHeader.jsx";
+import LeftPanel from "../components/leftPanel";
+import Button from "../components/button";
+import { useState, React, useEffect } from "react";
+import AddMemberWindow from "../components/addMemberWindow";
+import "../css/personalPage.css";
+import UploadNotesWindow from "../components/uploadNotesWindow";
+import { useParams } from "react-router";
+import SectionTitle from "../components/sectionTitle.jsx";
 
 function GroupPage(props) {
+    const [buttonAddMember, setButtonAddMember] = useState(false);
+    const [buttonGroupProfile, setButtonGroupProfile] = useState(false);
+    const [members, setMembers] = useState([]);
+    const [groups, setGroups] = useState([]);
+    const [notes, setNotes] = useState([]);
+    const [buttonUpload, setButtonUpload] = useState(false);
+    const [group, setGroup] = useState("");
+    const [groupDescription, setGroupDescription] = useState("");
+    const { groupId } = useParams();
+
+    const getGroupInfo = async () => {
+        let success = true;
+        const groupInfoResponse = await fetch(
+            "http://localhost:3000/v1/groups/" + groupId,
+            {
+                method: "GET",
+                headers: {
+                    Authorization:
+                        "bearer " + window.localStorage.getItem("authToken"),
+                },
+            }
+        ).catch((err) => {
+            console.error(err);
+            success = false;
+        });
+
+        if (success) {
+            const groupInfoResponseJSON = await groupInfoResponse.json();
+            if (groupInfoResponseJSON.group_id) {
+                setGroup(groupInfoResponseJSON.group_name);
+                setGroupDescription(groupInfoResponseJSON.description);
+            } else {
+                console.error("Could not get group information.");
+            }
+        }
+    };
+
+    const getGroupMembers = async () => {
+        let success = true;
+        const membersResponse = await fetch(
+            "http://localhost:3000/v1/groups/" + groupId + "/members",
+            {
+                method: "GET",
+                headers: {
+                    Authorization:
+                        "bearer " + window.localStorage.getItem("authToken"),
+                },
+            }
+        ).catch((err) => {
+            console.error(err);
+            success = false;
+        });
+
+        if (success) {
+            const memberResponseJSON = await membersResponse.json();
+            setMembers([]);
+            if (memberResponseJSON.users) {
+                for (const member of memberResponseJSON.users) {
+                    setMembers((arr) => [...arr, member]);
+                }
+            }
```

Comment:
Why are you reseting the members array when you are setting it later?
I think the for loop can be replaced with ``setMembers(memberResponseJSON.users)``

Reply:
If it is not reset, it will display the name of the member twice or more times.

Reply:
It should not, now that you replaced the for loop.
With the for loop, you were basically appending the members to an existing list.
With the new code, you are completely overwriting the members array.

Reply:
Removed it.